src/OutputWriters/output_construction.jl

```diff
@@ -16,6 +17,24 @@ end
 ##### Function output fallback
 #####
 
+function mask_field!(output)
+    field_masking_event = mask_immersed_field!(output)
+    wait(field_masking_event)
+end
+
+
+function _construct_output(output, grid, kwargs...)
+    if output isa AbstractOperation
+        output = Field(output)
+    end
+
+    if grid isa ImmersedBoundaryGrid
+        mask_field!(output)
+    end
```

Comment:
Since `construct_output` is only called when the output is constructed, the masking is only performed for the first iteration, rather than every time we ask for output.